action.yml

```diff
@@ -36,43 +36,107 @@ outputs:
+            echo "IS_V2=true" >> $GITHUB_ENV
+            echo "PRIORITY=v2" >> $GITHUB_ENV
+            echo "v2_token=${TOKEN}" >> $GITHUB_OUTPUT
+          else
+            echo "Token is in V1 format..."
+            echo "IS_V1=true" >> $GITHUB_ENV
+            echo "PRIORITY=v1" >> $GITHUB_ENV
+            echo "v1_token=${TOKEN}" >> $GITHUB_OUTPUT
+          fi
         fi
 
-    - name: Build and Publish to Oxygen V2
-      shell: 'bash'
-      id: 'oxygen-cli-action'
-      if: env.IS_V2 == 'true'
-      run: |
-        yarn global add @shopify/oxygen-cli@latest
-        URL_OUTPUT=$(oxygen-cli oxygen:deploy \
-          --path=${{ inputs.path }} \
-          --assetsFolder=${{ inputs.oxygen_client_dir }} \
-          --workerFolder=${{ inputs.oxygen_worker_dir }})
-        echo "url=$URL_OUTPUT" >> $GITHUB_OUTPUT
-      env:
-        OXYGEN_DEPLOYMENT_TOKEN: ${{ inputs.oxygen_deployment_token }}
-
-    - name: Build and Publish to Oxygen V1
+    - name: Build and Publish to Oxygen
       shell: 'bash'
       id: 'oxygenctl-action'
-      if: env.IS_V2 == 'false'
+      if: always()
       env:
         OXYGEN_ASSETS_DIR: ${{ inputs.oxygen_client_dir }}
         OXYGEN_BUILD_COMMAND: ${{ inputs.build_command }}
         OXYGEN_COMMIT_MESSAGE: ${{ inputs.commit_message || 'No commit message' }}
-        OXYGEN_DEPLOYMENT_TOKEN: ${{ inputs.oxygen_deployment_token }}
         OXYGEN_WORKFLOW_ID: "${{ github.run_id }}_${{ github.run_attempt }}"
         OXYGEN_WORKER_DIR: ${{ inputs.oxygen_worker_dir }}
       run: |
-        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}";
-        [[ -z "${{ inputs.commit_timestamp }}" ]] && export OXYGEN_COMMIT_TIMESTAMP=$(date -u +"%Y-%m-%dT%H:%M:%SZ") || export OXYGEN_COMMIT_TIMESTAMP=${{ inputs.commit_timestamp }};
-        echo "url=$($GITHUB_ACTION_PATH/oxygenctl deploy --health-check='${{ inputs.oxygen_health_check }}')" >> $GITHUB_OUTPUT
+        [[ -n "${{ inputs.path }}" ]] && cd "${{ inputs.path }}"
+
+        if [ "${{ env.IS_V2 }}" == "true" ]; then
+          echo "Deploying..."
+          mkdir oxygen_v2
+          yarn global add @shopify/oxygen-cli@latest
+          rsync -a ./* ./oxygen_v2/ --exclude 'oxygen_v2' --exclude 'node_modules'
+
+          oxygen_v2_command="oxygen-cli oxygen:deploy \
+                --path=${{ inputs.path }} \
+                --assetsFolder=${{ inputs.oxygen_client_dir }} \
+                --workerFolder=${{ inputs.oxygen_worker_dir }} \
+                --token=${{ steps.check_version.outputs.v2_token }}"
+
+          if [[ "${{ env.PRIORITY }}" == "v2" ]]; then
+            echo "v2_url=$(cd oxygen_v2 && ln -s ../node_modules node_modules && $oxygen_v2_command --buildCommand='${{ inputs.build_command }}')" >> v2_output &
+          else
+            echo "v2_url=$(cd oxygen_v2 && ln -s ../node_modules node_modules && $oxygen_v2_command --buildCommand='${{ inputs.build_command }}' &> /dev/null)" >> v2_output &
```

Comment:
That being said I think I can actually combine some of those commands together 👍🏻 . Will try a couple of things.

Reply:
> The output of the command is the URL on a happy path. But the key here is that the whole command runs in the background with the & at the end. So, the v2_output here is just capturing the result so that we can use it further down as the GH action step output

Right, I was just talking about the `else` case in this case 👍🏻